plugins/modules/ec2_instance.py

```diff
@@ -170,6 +171,7 @@
       - This field is deprecated and will be removed in a release after 2026-12-01, use O(network_interfaces) or O(network_interfaces_ids) instead.
       - Mutually exclusive with O(network_interfaces).
       - Mutually exclusive with O(network_interfaces_ids).
+      - If used along with `vpc_subnet_id`, `vpc_subnet_id` is used as a fallback to prevent errors when no subnet ID is found in the `network`.
```

Comment:
```suggestion
      - If used along with O("vpc_subnet_id"), O("vpc_subnet_id")  is used as a fallback to prevent errors when O("network.subnet_id") is not specified.
```